### PR TITLE
[Federation]Fix forgeting to close file

### PR DESCRIPTION
**Pull request by FengyunPan**
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: federation/plugin/pkg/admission/schedulingpolicy/admission_test.go [hunk 1]
```diff
@@ -343,6 +343,7 @@ func newControllerWithTestServer(f func(w http.ResponseWriter, r *http.Request),
 	if err != nil {
 		return nil, err
 	}
+	defer file.Close()
```
Comment (madhusudancs) [inline federation/plugin/pkg/admission/schedulingpolicy/admission_test.go]: The defer call a few lines above (https://github.com/kubernetes/kubernetes/pull/48013/files#diff-a277f7e642ed73686876c214453fab77R340) removes the file entirely. Is it really required to close it in a defer call that just executes before remove?

**Review by madhusudancs — COMMENTED** [review]
/lgtm
/approve

File: federation/plugin/pkg/admission/schedulingpolicy/admission_test.go (hunk 1 above)
Comment (FengyunPan) [inline federation/plugin/pkg/admission/schedulingpolicy/admission_test.go]: To avoid accidents, I think closing file after opening it more safe.

Comment (madhusudancs) [inline federation/plugin/pkg/admission/schedulingpolicy/admission_test.go]: @FengyunPan out of curiosity, what kind of accidents? In theory, I agree with what you are saying. But in this particular case, does it apply or matter?

**Pull request merged**